tools/gcc-bridge/runtime/src/main/java/org/renjin/gcc/runtime/Stdlib.java

```diff
@@ -79,6 +79,72 @@ public static int strcmp(Ptr x, Ptr y) {
     return strncmp(x, y, Integer.MAX_VALUE);
   }
 
+  /**
+   * The <code>strcasecmp()</code> function compares <code>x</code> and <code>y</code> without sensitivity to case.
+   * 
+   * All alphabetic characters in <code>x</code> and <code>y</code> are converted to lowercase before comparison.
+   * 
+   * @param x
+   * @param y
+   * @return a value indicating the relationship between the two strings, as follows:
+   *         <table>
+   *         <th>
+   *         <td>Value</td>
+   *         <td>Meaning</td></th>
+   *         <tr>
+   *         <td>Less than 0</td>
+   *         <td><code>x</code> less than <code>y</code></td>
+   *         </tr>
+   *         <tr>
+   *         <td>0</td>
+   *         <td><code>x</code> equivalent to <code>y</code></td>
+   *         </tr>
+   *         <tr>
+   *         <td>Greater than 0</td>
+   *         <td><code>x</code> greater than <code>y</code></td>
+   *         </tr>
+   *         </table>
+   */
+  public static int strcasecmp(Ptr x, Ptr y) {
+    for (int i = 0; i < Integer.MAX_VALUE; ++i) {
+      int bx = Character.toLowerCase(x.getByte(i));
+      int by = Character.toLowerCase(y.getByte(i));
+
+      if (bx < by) {
+        return -1;
+      } else if (bx > by) {
+        return 1;
+      }
+      if (bx == 0) {
+        break;
+      }
+    }
+    return 0;
+  }
+
+  public static Ptr strchr(Ptr string, int ch) {
+    final int offset = nullTerminatedString(string).indexOf(ch);
+    return new OffsetPtr(string, offset);
+  }
+
+  public static Ptr strrchr(Ptr string, int ch) {
+    final int offset = nullTerminatedString(string).lastIndexOf(ch);
+    return new OffsetPtr(string, offset);
+  }
+
+  public static Ptr strstr(Ptr string, Ptr searched) {
+    final int offset = nullTerminatedString(string).indexOf(nullTerminatedString(searched));
+    return new OffsetPtr(string, offset);
```

Comment:
You can also use `string.pointerPlus(offset)` here. In most cases `string` will be an instance of `BytePtr` which also includes an offset, so you'll avoid an extra layer of indirection.